From 1b07d00a15d6a96d1a36b6a284c4fd5f2e2fa383 Mon Sep 17 00:00:00 2001
Message-Id: <1b07d00a15d6a96d1a36b6a284c4fd5f2e2fa383.1648732882.git.geliang.tang@suse.com>
From: Daniel Xu <dxu@dxuuu.xyz>
Date: Mon, 23 Aug 2021 19:43:46 -0700
Subject: [PATCH] bpf: Add BTF_ID_LIST_GLOBAL_SINGLE macro
Git-commit: 1b07d00a15d6a96d1a36b6a284c4fd5f2e2fa383
Patch-mainline: v5.15-rc1
References: bsn#127
Modified-by-SEL: No


Same as BTF_ID_LIST_SINGLE macro except defines a global ID.

Signed-off-by: Daniel Xu <dxu@dxuuu.xyz>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Link: https://lore.kernel.org/bpf/a867a97517df42fd3953eeb5454402b57e74538f.1629772842.git.dxu@dxuuu.xyz
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 include/linux/btf_ids.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/btf_ids.h b/include/linux/btf_ids.h
index bed4b9964581..6d1395030616 100644
--- a/include/linux/btf_ids.h
+++ b/include/linux/btf_ids.h
@@ -82,6 +82,9 @@ __BTF_ID_LIST(name, globl)
 #define BTF_ID_LIST_SINGLE(name, prefix, typename)	\
 	BTF_ID_LIST(name) \
 	BTF_ID(prefix, typename)
+#define BTF_ID_LIST_GLOBAL_SINGLE(name, prefix, typename) \
+	BTF_ID_LIST_GLOBAL(name) \
+	BTF_ID(prefix, typename)
 
 /*
  * The BTF_ID_UNUSED macro defines 4 zero bytes.
-- 
2.34.1

